docs/community/tutorials-contest/2022/red-dev-ecrecover-x-chain/README.md

```diff
@@ -0,0 +1,362 @@
+# How to Extend Avalanche's Subnet-evm with a Stateful Precompile
+
+This tutorial will show you how to extend Avalanche's Subnet-evm with custom functionality. It adds a novel stateful precompile contract to verify the X-Chain signature of a message that has been signed using the Avalanche Wallet or by other means.
+
+Previously, we at [red·dev](https://www.red.dev) wrote a [tutorial](https://docs.avax.network/community/tutorials-contest/red-dev-sig-verify-tutorial) on the same topic using solidity. But we think it is faster, more user-friendly, and more gas-efficient to recreate it as a stateful precompile that works similarly to `ecrecover()`, a precompile that is already built into Subnet-evm that can verify C-Chain signatures.
+
+In this tutorial, we describe each step of setting up the environment and adding the new stateful precompile. For more information, see the **Resources** section at the end of this tutorial.
+
+## Audience
+
+To get the most out of this tutorial, you will need to have a basic understanding of Avalanche, Linux, Go (a.k.a. GoLang), Solidity, and how to write dApps and use Remix. If you do not yet know about these topics, see the [**Resources**](#resources) section at the end for links to learn more.
+
+# Prerequisites
+
+1. Ubuntu 20.04 or later
+2. [Go 1.18.3](https://storage.googleapis.com/golang/go1.18.3.linux-amd64.tar.gz)
+3. [Metamask wallet](https://metamask.io/) set up for use with Local EVM Subnet
+4. User with _sudo_ access
+
+## Install Go
+
+Download the Go package. We have used version 1.18.3 for this tutorial:
+```
+$ wget https://storage.googleapis.com/golang/go1.18.3.linux-amd64.tar.gz
+```
+Extract `go1.18.3.linux-amd64.tar.gz` to `/usr/local`:
+```
+$ sudo tar -C /usr/local -xzf go1.18.3.linux-amd64.tar.gz
+```
+Add `/usr/local/go/bin` to the PATH environment variable. You can do this by adding the following line to your `$HOME/.profile` or `/etc/profile` (for a system-wide installation):
+
+```
+export PATH=$PATH:/usr/local/go/bin
+```
+Verify that you've installed Go by running the command below:
+
+```
+$ go version
+```
+![go-v](ecrecover-x-chain-00-go-v.png)
+
+Set `$GOPATH` environment variable properly for Go to look for Go Workspaces. Please read [this](https://go.dev/doc/gopath_code) for details. You can verify by running `echo $GOPATH`.
+
+As some software will be installed into `$GOPATH/bin`, please make sure that `$GOPATH/bin` is in your `$PATH`. Otherwise, you may get errors running the commands below.
+
+## Clone Subnet-evm
```

Comment:
```suggestion
## Clone the Subnet EVM Repo
```

Reply:
@yulin-dong please advise on correct spelling for references to subnet-evm (https://github.com/ava-labs/subnet-evm)

Reply:
Subnet-EVM is used more frequently.